src/ExerciseList.hs

```diff
@@ -74,6 +74,7 @@ allExercises =
   , ExerciseInfo "Lists2" "lists" UnitTests "Your countdown doesn't need to end exactly with 0."
   , ExerciseInfo "Lists3" "lists" UnitTests "Remember 3 steps for a comprehension: 1. source 2. filter 3. result."
   , ExerciseInfo "Lists4" "lists" UnitTests "Remember 'show' and string appending!"
+  , ExerciseInfo "Curry1" "currying" CompileOnly "Use 'uncurry' and 'curry' and remember their type signatures."
```

Comment:
If we're moving this to the `functions` directory, we can just call it `Curry` until we add another exercise. Also I think we can reasonably have it as a `UnitTests` exercise. I'll recommend some in the wrap-up comment.